src/document.h

```diff
@@ -665,8 +667,11 @@ struct Document {
         DrawSelect(dc, selected);
         if (hover.g) hover.g->DrawHover(this, dc, hover);
         if (scaledviewingmode) { dc.SetUserScale(1, 1); }
-        if (scrolltoselection) {
+        if (initialzoomtoselection) {

```

Comment:
Why don't you just do this if `initialzoomlevel` is not 0 or -1 or whatever? Adding extra bools for each little thing gets very confusing, you want as little of them as possible :)

Reply:
Thanks for your valuable feedback! I revised the commit accordingly and I hope that it fits better now.

Reply:
> Why don't you just do this if `initialzoomlevel` is not 0 or -1 or whatever? Adding extra bools for each little thing gets very confusing, you want as little of them as possible :)

This a lot smarter! Thanks :-)